app/src/main/java/br/com/concrete/magic/domain/usecase/GetExpansionsUseCase.kt

```diff
@@ -1,15 +1,49 @@
 package br.com.concrete.magic.domain.usecase
 
+import br.com.concrete.magic.domain.model.Expansion
+import br.com.concrete.magic.domain.model.ExpansionView
+import br.com.concrete.magic.domain.model.ExpansionViewType
 import br.com.concrete.magic.domain.model.Root
 import br.com.concrete.magic.domain.repository.ExpansionsRepository
 import com.android.post.domain.usecase.base.UseCase
 
 class GetExpansionsUseCase constructor(
-    private val expansionsRepository: ExpansionsRepository
-) : UseCase<Root, Any?>() {
+        private val expansionsRepository: ExpansionsRepository
+) : UseCase<List<ExpansionView>, Any?>() {
 
-    override suspend fun run(params: Any?): Root {
-        return expansionsRepository.getExpansions()
+    override suspend fun run(params: Any?): List<ExpansionView> {
+        return formatExpansionList(sortExpansions(expansionsRepository.getExpansions()))
+//        prepareExpansionViews(
+//                sortExpansions(expansionsRepository.getExpansions())
+//        )
+    }
+
+    fun sortExpansions(root: Root): List<Expansion> {
+
+        val copy = arrayListOf<Expansion>().apply { addAll(root.sets) }
+        copy.sort()
+        root.sets = copy
+
+        return root.sets
```

Comment:
Essa ordenação poderia ser implementada de uma forma mais simples:
```suggestion
        return roots.sets.toMutableList().apply {
            sort()
        }
```